examples/ticks_and_spines/tick_labels_from_values.py

```diff
@@ -0,0 +1,37 @@
+"""
+
+Basic demo showing how to set tick labels to values of a series.
+
+Using ax.set_xticks causes the tick labels to be set on the currently
+chosen ticks. However, you may want to allow matplotlib to dynamically
+choose the number of ticks and their spacing.
+
+In this case may be better to determine the tick label from the value
+at the tick. The following example shows how to do this.
+
+NB: The MaxNLocator is used here to ensure that the tick
+values take integer values. As such, we need to catch
+any IndexErrors in the format function where we have not
+defined a label for that particular tick
+
+"""
+
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FuncFormatter, MaxNLocator
+fig = plt.figure()
+ax = fig.add_subplot(111)
+xs = range(26)
+ys = range(26)
+labels = list('abcdefghijklmnopqrstuvwxyz')
+
+
+def format_fn(tick_val, tick_pos):
+    try:
+        return labels[int(tick_val)]
```

Comment:
I would be happy to do that, however, I personally feel that demos need to be as simple as possible. They should efficiently communicate just one idea. If I continue to put in various fixes, the thing we are trying to exemplify becomes lost. It makes it more difficult for people to learn from the demos. 

Any problems encountered after having learnt this idea could be treated as a personal programming problem, or perhaps found in another demo somewhere. This depends on the problems encountered.

In fact, I would argue that many matplotlib demos fall into this trap. There is too much other code which detracts from the main idea. 

If you still feel this demo would be better with the negative integers fixed, I will fix it.